frontend/src/app/modules/workspace/settings/settings.component.ts

```diff
@@ -0,0 +1,202 @@
+import { HttpClient } from '@angular/common/http';
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import jwt_decode from 'jwt-decode';
+import { of, switchMap } from 'rxjs';
+import { GyazoUpload } from 'src/app/models/gyazo';
+import { GyazoService } from 'src/app/services/gyazo.service';
+import { environment } from 'src/environments/environment';
+
+@Component({
+  selector: 'app-settings',
+  templateUrl: './settings.component.html',
+  styleUrls: ['./settings.component.scss'],
+})
+export class SettingsComponent implements OnInit {
+  private readonly APIUrl = environment.apiUrl;
+  public settingsForm1: FormGroup = {} as FormGroup;
+  public settingsForm2: FormGroup = {} as FormGroup;
+  public avatar!: string;
+
+  public oldName: any = '';
+  public oldEmail: any = '';
+  public userId: any;
+  public name: any;
+  public imageFile!: File;
+
+  constructor(
+    private gyazoService: GyazoService,
+    private formBuilder: FormBuilder,
+    private httpService: HttpClient
+  ) {}
+
+  ngOnInit(): void {
+    this.initValues();
+    this.initForms();
+  }
+
+  initValues() {
+    let token = localStorage.getItem('accessToken') || '';
+
+    if (token) {
+      let decoded: any = jwt_decode(token);
+
+      this.name = decoded.name;
+      this.oldName = name;
+      this.oldEmail = decoded.email;
+      this.userId = decoded.id;
+    }
+  }
+
+  initForms() {
+    this.settingsForm1 = this.formBuilder.group({
+      workspaceName: [
+        this.name,
+        {
+          validators: [Validators.required, Validators.minLength(4)],
+        },
+      ],
+    });
```

Comment:
Validators for workspaceName:
Validators.pattern('^[a-zA-Z\`\'][a-zA-Z-\`\' ]+[a-zA-Z\`\']?$'),
          Validators.minLength(3),
          Validators.maxLength(30),